src/persist/src/sqlite.rs

```diff
@@ -0,0 +1,154 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! Implementation of [Consensus] backed by sqlite.
+
+use std::path::Path;
+use std::sync::Arc;
+use std::time::Instant;
+
+use async_trait::async_trait;
+use rusqlite::{named_params, params, Connection, OptionalExtension};
+use tokio::sync::Mutex;
+
+use crate::error::Error;
+use crate::storage::{Consensus, SeqNo, VersionedData};
+
+const APPLICATION_ID: i32 = 0x0678_ef32; // chosen randomly
+
+const SCHEMA: &str = "
+CREATE TABLE consensus (
```

Comment:
feels like this should have a primary key. perhaps `(shard, sequence_number)`?

Reply:
dunno about sqlite, you'd have to check, but in e.g. postgres if you don't specify a primary key, it invents a meaningless one. then your unique index ends up as a second map that essentially points at the meaningless (hidden) primary key. if you explictly declare a meaningful primary key, you can(i think) save the second map here

as for what the primary key is, for the current state where we're not yet keeping history, shard is a fine primary key. but when we add scan and truncate, i think we'll want the both the uniqueness constraint and pk to be (shard, seqno). if you insist, i suppose i'm okay with starting with the first one and changing to the second later, but given that a large part of this work is de-risking the eventual api, I think it's more useful to start with the second